class Solution {
    public int sumOfSquares(int[] nums) {
        
        int n1=nums.length;
        int sum=0;
        for(int i=0;i<nums.length;i++)
        {
            int n=i+1;
            if(n1%n==0)
            {
                sum +=(nums[i]*nums[i]);
            }
        }
        return sum;
    }
}